be/src/column/array_column.cpp

```diff
@@ -130,6 +130,18 @@ void ArrayColumn::append_default(size_t count) {
     _offsets->append_value_multiple_times(&offset, count);
 }
 
+void ArrayColumn::fill_default(const Filter& filter) {
+    std::vector<uint32_t> indexes;
+    for (size_t i = 0; i < filter.size(); i++) {
+        if (filter[i] == 1 && get_element_size(i) > 0) {
+            indexes.push_back(i);
+        }
+    }
+    auto default_column = clone_empty();
+    default_column->append_default(indexes.size());
+    update_rows(*default_column, indexes.data());
```

Comment:
update_rows is quite a time-consuming function, it would be better if we can use memmove to compact offset/null columns and primitive column.

Reply:
In most case we don't need to really `update_null_with_default`, we could skip this procedure if the null element are already default.